fontbe/src/metrics_and_limits.rs

```diff
@@ -0,0 +1,204 @@
+//! Generates the [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx) and
+//! [hhea](https://learn.microsoft.com/en-us/typography/opentype/spec/hhea) tables
+
+use std::cmp::{max, min};
+
+use fontdrasil::orchestration::Work;
+use read_fonts::types::FWord;
+use write_fonts::{
+    dump_table,
+    tables::{hhea::Hhea, hmtx::Hmtx, vmtx::LongMetric},
+    OtRound,
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, Glyph},
+};
+
+struct MetricAndLimitWork {}
+
+pub fn create_metric_and_limit_work() -> Box<BeWork> {
+    Box::new(MetricAndLimitWork {})
+}
+
+#[derive(Debug, Default)]
+struct GlyphLimits {
+    min_left_side_bearing: Option<i16>,
+    min_right_side_bearing: Option<u16>,
+    x_max_extent: Option<i16>,
+    advance_width_max: Option<u16>,
+    max_points: Option<u16>,
+    max_contours: Option<u16>,
```

Comment:
maybe a lot of these fields don't need to be `Option`? would clear up a bunch of the other code...

Reply:
I think you are right. I made them option to match maxp but we would branch at a higher level if we were trying to emit a maxp v0.5 and they are mandatory for 1.0.

Reply:
Fixed.